prusti-specs/src/export_spec_rewriter.rs

```diff
@@ -0,0 +1,334 @@
+        if let Some(export_path_attr) = $item.attrs.iter().find(|attr| {
+            attr.path.segments.len() == 2
+                && &attr.path.segments.last().unwrap().ident.to_string() == $attr_str
+        }) {
+            for tt in export_path_attr.tokens.to_owned() {
+                if let TokenTree::Group(group) = tt {
+                    $path_mut_ref = syn::parse2(group.stream()).unwrap();
+                }
+            }
+        }
+    };
+}
+
+// expand the module path recorded when traversing AST
+fn expand_mod_path_in_macro(
+    path: &syn::Path,
+    mut tokens: TokenStream,
+    uses_same_level: &TokenStream,
+) -> TokenStream {
+    match &path.segments.empty_or_trailing() {
+        true => tokens,
+        false => {
+            for seg in path.segments.to_owned().into_iter().rev() {
+                tokens = quote!(
+                    mod #seg {
+                        #uses_same_level
+                        #tokens
+                    }
+                );
+            }
+            tokens
+        }
+    }
+}
+
+fn expand_type_path_in_macro(
+    path: &syn::Path,
+    tokens: TokenStream,
+    uses_same_level: &TokenStream,
+    type_path: &syn::Path,
+) -> TokenStream {
+    let type_path_macro_ident = generate_macro_ident(type_path.to_owned());
+    match type_path.segments.empty_or_trailing() {
+        true => expand_mod_path_in_macro(path, tokens, uses_same_level),
+        false => expand_mod_path_in_macro(
+            path,
+            quote!(
+                $crate::#type_path_macro_ident!();
+                #tokens
+            ),
+            uses_same_level,
+        ),
+    }
+}
+
+pub fn rewrite_export_spec(path: &syn::Path, item: &syn::Item) -> syn::Result<TokenStream> {
+    let mut macros_tokens = TokenStream::new();
+    let mod_path = syn::Path {
+        leading_colon: None,
+        segments: syn::punctuated::Punctuated::new(),
+    };
+    match item {
+        syn::Item::Fn(item_fn) => {
+            rewrite_fn(
+                path,
+                mod_path,
+                item_fn,
+                &mut macros_tokens,
+                &TokenStream::new(),
+            );
+        }
+        syn::Item::Mod(item_mod) => {
+            rewrite_mod(
+                path,
+                mod_path,
+                item_mod,
+                &mut macros_tokens,
+                &TokenStream::new(),
+            )?;
+        }
+        syn::Item::Impl(item_impl) => {
+            rewrite_impl(
+                path,
+                mod_path,
+                item_impl,
+                &mut macros_tokens,
+                &TokenStream::new(),
+            )?;
+        }
+        syn::Item::Struct(item_struct) => {
+            rewrite_struct(
+                item_struct,
+                mod_path,
+                &mut macros_tokens,
+                &TokenStream::new(),
+            );
+        }
+        _ => {
+            unimplemented!();
```

Comment:
Is this panic actually reachable if the user uses Prusti badly? If yes, there should be an explanation here. If not, this should be `unreachable!()` instead.